docs/adr/0007-rancher-turtles-public-api.md

```diff
@@ -0,0 +1,75 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [7. Rancher Turtles Public API](#7-rancher-turtles-public-api)
+  - [Context](#context)
+  - [Decision](#decision)
+  - [Consequences](#consequences)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# 7. Rancher Turtles Public API
+
+* Status: accepted
+* Date: 2023-12-04
+* Authors: @Danil-Grigorev @richardcase @alexander-demicev
+* Deciders: @richardcase @alexander-demicev @furkatgofurov7 @Danil-Grigorev @mjura
+
+## Context
+
+Rancher Turtles requires a public API implementation to facilitate provisioning and operations over `CAPI Operator` and `Cluster API` resources.
+
+This resource should:
+
+1. Support templating `CAPI Operator` `*Provider` manifests, including the required resource for the air-gapped environments, manager configurations and environment secrets for the provider.
+1. Map `Rancher` credentials content to the `CAPI Operator` credentials secret.
+1. Allow GitOps compatibility and reusability.
+
+## Decision
+
+The decided implementation of the API resource will be located under the `turtles-capi.cattle.io` group, kind `CAPIProvider`.
+
+Example specification of the resource:
+```yaml
+apiVersion: turtles-capi.cattle.io/v1alpha1
+kind: CAPIProvider
+metadata:
+  name: aws-infra
+  namespace: default # Resource is namespace scoped.
+spec:
+  name: aws # Defines the provider kind to provision.
+  type: infrastructure # CAPI Provider resource kind to provision.
+  variables: # Additional environment variables to be declared inside the referenced spec.configSecret
+    CAPA_LOGLEVEL: "4"
+  configSecret:
+    name: aws-credentials # This secret will be created or adjusted based on the content of the spec.features and spec.variables, without overriding non-overlapping content.
+  credentials:
+    rancherCloudCredential: user-credential # Rancher secret content wll be translated to CAPI Operator environment secret.
```

Comment:
Q: trying to understand here, why do we need to refer rancher cc and what is the benefit?

Reply:
This will allow to translate rancher credentials to CAPI credentials, which is also called credentials mapper. In UI workflow this would mean a user can use existing rancher credentials to create a CAPI cluster, without a need to create a separate CAPI provider secret.

Reply:
I see, AFAIR there was a separate issue to track credentials mapping but fine for me to already having that in mind in this PR